File: frontend/figma-make-code/Online Calendar Application/types/meeting.ts
export interface Participant {
  id: string;
  email: string;
  name: string;
  notificationChannels: {
    email: boolean;
    push: boolean;
  };
  response?: 'yes' | 'no' | 'pending'; // 参加者の回答状況を追加
}

export interface Meeting {
  id: string;
  title: string;
  startTime: Date;
  endTime: Date;
  owner: string;
  participants: Participant[];
  isImportant: boolean;
  status: 'scheduled' | 'in-progress' | 'completed' | 'cancelled';
  createdAt: Date;
  updatedAt: Date;
}

export interface MeetingStats {
  totalMeetingsOwned: number;
  totalMeetingsParticipated: number;
  averageDailyMinutes: number;
  weeklyData: {
    date: string;
    dayName: string;
    totalMinutes: number;
  }[];
}

File: ios/Sources/Core/Meeting/Models/Meeting.swift
import Foundation

struct Meeting: Identifiable {
    let id: String
    let title: String
    let description: String?
    let startDate: Date
    let endDate: Date
    let organizer: Organizer
    let participants: [Participant]
    
    // 通常のイニシャライザ
    init(
        id: String,
        title: String,
        description: String? = nil,
        startDate: Date,
        endDate: Date,
        organizer: Organizer,
        participants: [Participant] = []
    ) {
        self.id = id
        self.title = title
        self.description = description
        self.startDate = startDate
        self.endDate = endDate
        self.organizer = organizer
        self.participants = participants
    }
    
    // APIレスポンスからの変換用
    init(from response: MeetingResponse) {
        self.id = response.id
        self.title = response.title
        self.description = nil
        self.startDate = response.startTime
        self.endDate = response.endTime
        self.organizer = Organizer(
            id: response.ownerId,
            name: response.owner.components(separatedBy: "@").first ?? response.owner,
            email: response.owner
        )
        self.participants = response.participants.map { simpleParticipant in
            Participant(
                id: simpleParticipant.id,
                email: simpleParticipant.email,
                name: simpleParticipant.name
            )
        }
    }
}

struct Organizer: Codable {
    let id: String
    let name: String
    let email: String
}

struct Participant: Codable {
    let id: String
    let email: String
    let name: String
}